src/oumi/inference/gcp_inference_engine.py

```
@@ -27,6 +28,92 @@
+        "{project_id}/locations/{region}/endpoints/openapi/chat/completions"
+    )
+    """The API URL template for the GCP project. Used when no `api_url` is provided."""
+
+    _DEFAULT_PROJECT_ID_ENV_KEY: str = "PROJECT_ID"
+    """The default project ID environment key for the GCP project."""
+
+    _DEFAULT_REGION_ENV_KEY: str = "REGION"
+    """The default region environment key for the GCP project."""
+
+    _project_id: Optional[str] = None
+    """The project ID for the GCP project."""
+
+    _region: Optional[str] = None
+    """The region for the GCP project."""
+
+    def __init__(
+        self,
+        model_params: ModelParams,
+        *,
+        generation_params: Optional[GenerationParams] = None,
+        remote_params: Optional[RemoteParams] = None,
+        project_id_env_key: Optional[str] = None,
+        region_env_key: Optional[str] = None,
+        project_id: Optional[str] = None,
+        region: Optional[str] = None,
+    ):
+        """Initializes the inference Engine.
+
+        Args:
+            model_params: The model parameters to use for inference.
+            generation_params: The generation parameters to use for inference.
+            remote_params: The remote parameters to use for inference.
+            project_id_env_key: The environment variable key name for the project ID.
+            region_env_key: The environment variable key name for the region.
+            project_id: The project ID to use for inference.
+            region: The region to use for inference.
+        """
+        super().__init__(
+            model_params=model_params,
+            generation_params=generation_params,
+            remote_params=remote_params,
+        )
+        self._project_id_env_key = (
+            project_id_env_key or self._DEFAULT_PROJECT_ID_ENV_KEY
+        )
+        self._region_env_key = region_env_key or self._DEFAULT_REGION_ENV_KEY
+        self._project_id = project_id
+        self._region = region
```

We should probably return an error if the user supplies both `project_id_env_key` and `project_id`, or both `region_env_key` and `region`